### DISCUSSION TITLE
`compile_commands.json` issues

### DISCUSSION THREAD

**Discussion by Natan-P** [davatorium/rofi#1959] [Q&A]
I generated the compile commands file by running meson's build system. After setting ccls to look for it at the correct location, I noticed that `config.h` was being reported as not found. I noticed that it is only made during compilation, so added it to a directory and added an include flag to it in `.ccls`. By doing this, `rofi.c` also gave an error about `nkutils-git-version.h` not being found, line 50 in `rofi.c` being the issue. This is caused by config.h having `USE_NK_GIT_VERSION` defined, but the .h file in question is generated during compilation aswell, I assume.
This is not the intended way to get IDE support, right?
 
I also noticed that `resources/resources.h` was not being included - some quick searching through the (attached) compile_commands.json file lead me to believe it isn't included.
I haven't tried generating the file through `bear` yet, only meson.

What programs do you use for IDE support/what am I doing wrong here?

(in case it is needed, here is the [compile_commands.json](https://github.com/davatorium/rofi/files/14551641/compile_commands.json) file meson outputs)

**Comment by DaveDavenport** [comment]
I am a bit confused what the problem is?  I assume rofi itself compiles fine both with meson and autoconf (this is checked in CI).

It sounds like the files you have problems with are generated files, they normally exist in the build directory. `resources.h` is used by the lexer, this is also a generated file (from lexer/theme-lexer.l). 

However I am not sure what you are trying todo?  what is `.ccls`?



I use nvim + lazyvim as IDE and that works fine.

**Reply by Natan-P** [reply]
Hi, I'm sorry. I tried sanity checking my actions, so I re-downloaded the source code and built it. Everything worked.
I think that in my previous antics after stumbling on trying to include everything by hand by reading out the Makefile, I nuked build/, only ran `meson build` and didn't actually compile, which resulted in some missing files. 

So, for anyone stumbling upon this thread in the future, properly build the thing (including compilation) before trying to develop stuff.
In terms of C and the related toolchain, I've only ever worked on small projects consisting of a couple files and a tiny Makefile for the build system, so most of this wasn't known to me.
Thanks for your quick reply!
